src/class/number.rs

```diff
@@ -647,22 +647,22 @@ where
 {
     fn tp_as_number() -> Option<ffi::PyNumberMethods> {
         Some(ffi::PyNumberMethods {
-            nb_add: Self::nb_add(),
-            nb_subtract: Self::nb_subtract(),
-            nb_multiply: Self::nb_multiply(),
+            nb_add: Self::nb_add().or_else(Self::nb_add_fallback),
```

Comment:
So I haven't followed the rest of the logic in too much detail, but I have a question here.

If a `#[pyclass]` defines both `__add__` and `__radd__`, it looks like `__add__` wins here. What happens to `__radd__` - can it still be called?

Reply:
> If a #[pyclass] defines both __add__ and __radd__, it looks like __add__ wins here

Yeah, but this is the same as what CPython does.
```python
In [4]: class C: 
   ...:     def __add__(self, other): 
   ...:         return f"Self + {other}" 
   ...:     def __radd__(self, other): 
   ...:         return f"{other} + Self" 
   ...:     def __repr__(self): 
   ...:         return "C" 
   ...:                                                                         

In [5]: c = C()                                                                 

In [6]: c + c                                                                   
Out[6]: 'Self + C'
```

Reply:
That makes sense. What I mean is should we add a test for a class in `test_arithmetic` which defines both `__add__` and `__radd__`, and check everything works as expected? 

E.g. `c + c -> __add__` , `c + 1 -> __add__`, `1 + c -> __radd__` .